owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt

```diff
@@ -209,20 +225,32 @@ class DocumentsStorageProvider : DocumentsProvider() {
         Log_OC.d(TAG, "Trying to delete ${file.fileName} with id ${file.fileId}")
 
         RemoveFileOperation(file.remotePath, false).apply {
-            execute(currentStorageManager, context).also { checkOperationResult(it, file) }
+            execute(currentStorageManager, context).also { checkOperationResult(it, file.parentId.toString()) }
         }
     }
 
-    private fun checkOperationResult(result: RemoteOperationResult<Any>, file: OCFile) {
+    private fun checkOperationResult(result: RemoteOperationResult<Any>, folderToNotify: String) {
         if (!result.isSuccess) {
-            notifyChangeInFolder(file.parentId.toString())
+            if (result.code != RemoteOperationResult.ResultCode.WRONG_CONNECTION) notifyChangeInFolder(folderToNotify)
             throw FileNotFoundException("Remote Operation failed due to ${result.exception.message}")
         } else {
```

Comment:
You might not need the else condition since you are throwing an exception just before and the code below else will be only reachable when the result is successes.

Reply:
Done